packages/chord-detect/index.ts

```diff
@@ -33,12 +37,41 @@ export function detect(source: string[]): string[] {
     .map((chord) => chord.name);
 }
 
+// Assumes that chord is presorted
+function findRoot(chord: string[]): string {
+  let foundRoot = null;
+  chord.every((note) => {
+    const workComplete = chord.some((otherNote) => {
+      const interval = simplify(distance(note, otherNote));
+      const orderedNotes = sortedNames([note, otherNote]);
+      if (interval === "5P") {
+        foundRoot = orderedNotes[0];
+        return foundRoot; // Loop is complete
+      } else if (interval === "4P") {
+        foundRoot = orderedNotes[1];
+        return foundRoot; // Loop is complete
+      }
+      return false; // continue looping
+    });
+    // Continue looping if the root note was not found
+    return !workComplete;
+  });
+  if (foundRoot) {
+    return foundRoot;
+  } else {
+    // Defaults to the old behavior if the chord is complex and the root note cannot be easily found
+    return chord[0];
+  }
+}
+
+// assumes that notes is presorted
 function findExactMatches(notes: string[], weight: number): FoundChord[] {
-  const tonic = notes[0];
-  const tonicChroma = note(tonic).chroma;
+  const root = findRoot(notes);
```

Comment:
Is this name change on purpose? Can you elaborate the reasoning?

Reply:
This was changed to root since it is looking for a root of a chord and not the tonic of a scale.